### DISCUSSION TITLE
Double tap key binding?

### DISCUSSION THREAD

**Discussion by dromoli** [orbitalquark/textadept#223] [Q&A]
Is there a way to assign a command to "Alt+Alt" (double tap key binding)?

**Comment by orbitalquark** [comment]
No, there is not. You could attempt some trickery involving `events.KEYPRESS`, but I don't know how successful you'd be.

**Comment by mhwombat** [comment]
I tried this just now and it worked. The print statement is just there temporarily so you can see the code for each key that is pressed, in case your Alt key is a different code, or you want to map both Alt keys.

```
local have_first_alt = false

events.connect(events.KEYPRESS, function(code, shift, control, alt, cmd, caps)
  print(code, keys.KEYSYMS[code], shift, control, alt, cmd, caps)
  if code==65513 then
    if have_first_alt then
      print('I saw two alts in a row')
      return true -- I've handled it, no need to propagate this event
    else
      have_first_alt = true
    end
  else
    have_first_alt = false
  end
  return
end, 1)

```

**Comment by orbitalquark** [comment]
That's the kind of trickery I had in mind. Thanks for coming up with a working example :)

**Comment by dromoli** [comment]
Thanks. I also experimented a little and came up with this:

```
events.connect(events.KEYPRESS, function(code)
    if code == 65514 then
--        ui.print('Initial time: ' .. initialAltTime)
        if initialAltTime ~= 0 then
            finalAltTime = os.time()
--            ui.print('Final time: ' .. finalAltTime)
            if (finalAltTime - initialAltTime) < 1 then
                ui.print('Less than a second between Alt keypresses')
                resetAltTimes()
            else
                resetAltTimes()
                return
            end
        else
            initialAltTime = os.time()
        end
    end
end)

function resetAltTimes()
    initialAltTime = 0
    finalAltTime = 0
end
```

It tries to discard the event if the time elapsed between keypresses is more than one second (to avoid reacting every second time the Alt key is pressed and only doing it if chances are this was really a double tap).

It can still be improved.
